packages/playwright-core/src/server/screenshotter.ts

```diff
@@ -62,79 +62,50 @@ export class Screenshotter {
   async screenshotPage(progress: Progress, options: types.ScreenshotOptions): Promise<Buffer> {
     const format = validateScreenshotOptions(options);
     return this._queue.postTask(async () => {
-      const { viewportSize, originalViewportSize } = await this._originalViewportSize(progress);
+      const { viewportSize } = await this._originalViewportSize(progress);
 
+      const fullPageSize = await this._fullPageSize(progress);
```

Comment:
Let's keep existing logic and only call _fullPageSize under the options.fullPage branch.